### PR TITLE
Added News Details Activity

### PR DESCRIPTION
**Pull request by mshariqaftab**
- Added Sections in ListView 
- Added NewDetails Activity Class

### CODE REVIEW

File: app/src/main/res/layout/news_feed_list_row.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
         android:layout_height="wrap_content" />
 
     <TextView
-        android:id="@+id/title"
+        android:id="@+id/newsTitle"
```
Comment (praveen-garg) [inline app/src/main/res/layout/news_feed_list_row.xml:23]: why this fuck is not news_title?

File: app/src/main/res/layout/section.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<TextView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="48dp"
+    android:gravity="center_vertical"
+    android:paddingLeft="16dp"
+    android:maxLines="1"
+    android:textAllCaps="true"
+    android:textColor="@color/colorPrimaryDark"
```
Comment (praveen-garg) [inline app/src/main/res/layout/section.xml:10]: should it be color_primary_dark?

File: app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        int sectionedPosition;
+        CharSequence title;
+
+        public Section(int firstPosition, CharSequence title) {
+            this.firstPosition = firstPosition;
+            this.title = title;
+        }
+
+        public CharSequence getTitle() {
+            return title;
```
Comment (praveen-garg) [inline app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java:119]: should not this be `this.title`?

File: app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        }
+    }
+
+
+    public void setSections(Section[] sections) {
+        mSections.clear();
+
+        Arrays.sort(sections, new Comparator<Section>() {
+            @Override
+            public int compare(Section o, Section o1) {
```
Comment (praveen-garg) [inline app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java:129]: can we please be sensible here? and use proper names? WTF is o and o1?

File: app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+            ++offset;
+        }
+
+        notifyDataSetChanged();
+    }
+
+    public int positionToSectionedPosition(int position) {
+        int offset = 0;
+        for (int i = 0; i < mSections.size(); i++) {
+            if (mSections.valueAt(i).firstPosition > position) {
```
Comment (praveen-garg) [inline app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java:149]: anti-pattern?

File: app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+    public int sectionedPositionToPosition(int sectionedPosition) {
+        if (isSectionHeaderPosition(sectionedPosition)) {
+            return RecyclerView.NO_POSITION;
+        }
+
+        int offset = 0;
+        for (int i = 0; i < mSections.size(); i++) {
+            if (mSections.valueAt(i).sectionedPosition > sectionedPosition) {
+                break;
```
Comment (praveen-garg) [inline app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java:165]: anti-pattern?

File: app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java (hunk 3 above)
Comment (mshariqaftab) [reply app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java:119]: return title is correct, there is no need of _this_ here. we are already setting the value of title inside constructor.

Comment (praveen-garg) [reply app/src/main/java/com/alif/newsreader/adapter/SimpleSectionedRecyclerViewAdapter.java:119]: ugh! I was 🏃 through and misunderstood for setter, where we use 'this' to be explicit (instance vs function variable).

**Pull request merged**